packages/tizen_app_manager/lib/src/app_manager.dart

```diff
@@ -0,0 +1,261 @@
+  }
+
+  void destroy() {
+    if (_handle != nullptr) {
+      final ret = appManager.destroyContext(_handle);
+      _handle = nullptr;
+      if (ret != 0) {
+        throw PlatformException(
+          message: 'Failed to excute app_context_destroy',
+          code: _getErrorCode(ret),
+        );
+      }
+    }
+  }
+
+  int getHandleAddress() {
+    return _handleAddress;
+  }
+
+  bool isAppRunning() {
+    var isRun = false;
+    final Pointer<Int8> out = malloc();
+    final appIdP = _appId.toNativeUtf8();
+    try {
+      final ret = appManager.appIsRunning(appIdP, out);
+      if (ret != 0) {
+        throw PlatformException(
+          message: 'Failed to excute app_manager_is_running $_appId',
+          code: _getErrorCode(ret),
+        );
+      }
+
+      if (out != nullptr) {
+        isRun = out.value != 0 ? true : false;
+      }
+      return isRun;
+    } finally {
+      malloc.free(appIdP);
+      malloc.free(out);
+    }
+  }
+
+  String getPackageId() {
+    final Pointer<Pointer<Utf8>> packageIdP = malloc();
+    try {
+      _handleCheck();
+      final ret = appManager.getPackageId(_handle, packageIdP);
+      if (ret != 0) {
+        throw PlatformException(
+          message: 'Failed to excute app_context_get_package_id',
+          code: _getErrorCode(ret),
+        );
+      }
+
+      return packageIdP.value.toDartString();
+    } finally {
+      malloc.free(packageIdP);
+    }
+  }
+
+  int getProcessId() {
+    final Pointer<Int32> processIdP = malloc();
+    try {
+      _handleCheck();
+      final ret = appManager.getProcessId(_handle, processIdP);
+      if (ret != 0) {
+        throw PlatformException(
+          message: 'Failed to excute app_context_get_pid',
+          code: _getErrorCode(ret),
+        );
+      }
+      return processIdP.value;
+    } finally {
+      malloc.free(processIdP);
+    }
+  }
+
+  int getAppState() {
+    final Pointer<Int32> stateP = malloc();
+    try {
+      _handleCheck();
+      final ret = appManager.getAppState(_handle, stateP);
+      if (ret != 0) {
+        throw PlatformException(
+          message: 'Failed to excute app_context_get_app_state',
+          code: _getErrorCode(ret),
+        );
+      }
+      return stateP.value;
+    } finally {
+      malloc.free(stateP);
+    }
+  }
+
+  void terminate() {
+    _handleCheck();
+    final ret = appManager.terminateApp(_handle);
+    if (ret != 0) {
+      throw PlatformException(
+        message: 'Failed to excute app_manager_terminate_app',
+        code: _getErrorCode(ret),
+      );
+    }
+  }
+
+  void resume() {
+    _handleCheck();
+    final ret = appManager.resumeApp(_handle);
+    if (ret != 0) {
+      throw PlatformException(
+        message: 'Failed to excute app_manager_resume_app',
+        code: _getErrorCode(ret),
+      );
+    }
+  }
+
+  void _handleCheck() {
+    if (_handle == nullptr) {
+      throw PlatformException(
+          code: _getErrorCode(tizenErrorInvalidParameter),
```

Comment:
`PlatformException` doesn't seem to make sense. How about using `StateError` instead?

```dart
if (_handle == nullptr) {
  throw StateError('The object has been destroyed');
}
```